src/Network/WebSockets/Connection.hs

```diff
@@ -212,23 +225,32 @@ data ConnectionOptions = ConnectionOptions
     { connectionOnPong :: !(IO ())
       -- ^ Whenever a 'pong' is received, this IO action is executed. It can be
       -- used to tickle connections or fire missiles.
+    , connectionPermessageDeflate :: Maybe PermessageDeflate
+      -- ^ Settings of permessage deflate extension.
+      -- Nothing will disable it
     }
 
 
 --------------------------------------------------------------------------------
 defaultConnectionOptions :: ConnectionOptions
 defaultConnectionOptions = ConnectionOptions
     { connectionOnPong = return ()
+    , connectionPermessageDeflate = Just defaultPermessageDeflate
```

Comment:
Will need to remove this setting before the final merge.
It changes the memory usage of the library, from default nothing to +320k per connection.
I hacked it here for now, as there is no obvious support in yesod libraries I use.
But I can see it being highly dependent on the use case. 
Eg. we gradually downgrade memory usage for any new connection increasing bandwidth use (in my case compression rations are 10x-40x)